### ISSUE TITLE
Visualization is not working on Samsung n130

### ISSUE THREAD

**Issue by ghostbuster2** [Denvi/Candle#99]
Do not work on Windows 10/8.1/7
Work fine on ubuntu 16.10
Here is screenshots https://yadi.sk/d/zpQ9roQw34Ec9Y

**Comment by ghostbuster2** [comment]
В нетбуке установлена встроенная видеокарта семейства Mobile Intel(R) 945 Express Chipset. В версии программы из поста #64 визуализация завелась, но хотелось бы все-таки рабочую последнюю версию. Спасибо!

**Comment by ghostbuster2** [comment]
0.9. версия работает, подозреваю потому-что в ней есть все эти методы рендеринга
"Visualizer works with OpenGL/Angle/Software renderer"
Нужна по возможности версия 1.1 т.к. в контроллер залита последняя прошивка grbl

**Comment by Denvi** [comment]
Нужно собирать на Qt 5.5, на днях сделаю.

**Comment by ghostbuster2** [comment]
Ок, спасибо!

**Comment by Denvi** [comment]
[Candle_1.1.7_.99.zip](https://github.com/Denvi/Candle/releases/download/v1.1/Candle_1.1.7_.99.zip)

**Comment by ghostbuster2** [comment]
Благодарю за оперативность, все работает.

**Comment by Denvi** [comment]
> Do not work on Windows 10/8.1/7

Про XP речи не было... придется жить с ошибкой.

**Comment by ghostbuster2** [comment]
Если на функционал не влияет, то можно жить,😀

**Comment by Denvi** [comment]
Теперь не должен сыпать ошибки, если [редиску](https://www.microsoft.com/ru-ru/download/details.aspx?id=30679) установить.

[Candle_1.1.7.zip](https://github.com/Denvi/Candle/releases/download/v1.1/Candle_1.1.7.zip)

**Comment by ghostbuster2** [comment]
Значит сейчас расклад такой:
До установки "Распространяемый пакет Visual C++ для Visual Studio 2012 Обновление 4" по вашей ссылке выдавало такую ошибку
![error_2](https://cloud.githubusercontent.com/assets/16155546/21526873/65f2f064-cd38-11e6-8058-63745eff2399.PNG)
После установки - ситуация не изменилась, ошибка все та же.
Потом решил установить этот [пакет](https://www.microsoft.com/RU-ru/download/details.aspx?id=40784) После его установки ошибка сменилась на такую
![error_3](https://cloud.githubusercontent.com/assets/16155546/21526916/aad0d688-cd38-11e6-9268-12de017f7233.PNG)

**Comment by Denvi** [comment]
Ясно, буду искать тачку на древнем чипсете, но что-то мне подсказывает - бесполезно возиться с XP.

**Comment by ghostbuster2** [comment]
Вот и я к этому склоняюсь, верну наверное восьмерку и горя знать не буду. Спасибо за приложенные усилия.

**Comment by prulitarit** [comment]
> Ясно, буду искать тачку на древнем чипсете, но что-то мне подсказывает - бесполезно возиться с XP.

I successfully reproduced error, ghostbuster2 talking about,  in the virtualbox windows xp:

Without any vcredist installed:  - msvcp120.dll and msvcr120.dll not found error.
With vcredir 2013 installed - strnlen not found in msvcr.dll error

if I remove d3dcompiler_47.dll file, strnlen error goes away.

you can use virtualbox. it can emulate old chipsets.

**Comment by Denvi** [comment]
Yes, i have reproduced this error on virtualbox (but i can't simulate OpenGL 1.0 on it while my card was 4+) and on hardware. Anyway WindowsXP doesn't works with ANGLE, you should install newer graphics card.

**Comment by prulitarit** [comment]
I can not get newest graphics card because only Intel 945GM internal video chipset in this laptop. 

> Graphics card with OpenGL 2.0 support

My latest OpenGL version is 1.4 and Candle seems to work fine, (exept this error at startup). 
How can I determine actual rendering mode (OpenGL/Angle/Software renderer) ?
And what difference between these modes?
